crypto/cms/cms_ec.c

```diff
@@ -358,7 +362,7 @@ static int ecdh_cms_encrypt(CMS_RecipientInfo *ri)
      * of another AlgorithmIdentifier.
      */
     penclen = i2d_X509_ALGOR(wrap_alg, &penc);
-    if (penc == NULL || penclen == 0)
+    if (penc == NULL || penclen <= 0)
```

Comment:
The `NULL` check is redundant (see `asn1_item_flags_i2d()`) and the same strange check appears in `cms_dh.c`, so if you change it here, it should probably be changed there, too.

Reply:
OK